File: 프로그래머스/2/148652. 유사 칸토어 비트열/유사 칸토어 비트열.py
def check(x, n, target):
    for i in range(n, 0, -1):
        x, mod = divmod(x, 5)
        if mod == 2:
            return False
        if x == 1:
            return target[mod]
    return True


def solution(n, l, r):
    target = 1, 1, 0, 1, 1
    return sum(check(x, n, target) for x in range(l - 1, r))

File: 프로그래머스/2/148652. 유사 칸토어 비트열/유사 칸토어 비트열.java
class Solution {

    public boolean check(long x, int n, int[] target) {
        for (int i = n; i > 0; i--) {
            int mod = (int)(x % 5);
            x /= 5;

            if (mod == 2) {
                return false;
            }

            if (x == 1) {
                return target[mod] == 1;
            }
        }
        return true;
    }

    public int solution(int n, long l, long r) {
        int[] target = {1, 1, 0, 1, 1};
        int answer = 0;

        for (long x = l - 1; x < r; x++) {
            if (check(x, n, target)) {
                answer++;
            }
        }

        return answer;
    }
}